### ISSUE TITLE
There were many lost events in the trace

### ISSUE THREAD

**Issue by SudhindraChegu** [microsoft/perfview#457]
When I am running PerfView in Debug/Release mode, I am getting below mentioned error popup. It is consistent across multiple trials.

![image](https://user-images.githubusercontent.com/34036727/33466525-91674f88-d674-11e7-8580-488c3abe658a.png)

During Debug, I get an additional popup as shown below ::
![image](https://user-images.githubusercontent.com/34036727/33466551-b1bfbb62-d674-11e7-8c78-6c2cebbca64e.png)

**Comment by vancem** [comment]
The two issues are not likely to be related.    For the lost events, the possibility for lost events built into ETW, and the advice that is given (right now /InMemoryCircularBuffer I think might be broken, but the /BufferSize advice is reasonable.   

To diagnose  the assert, we need to at least know the assert that is firing (my code is different than yours as there is no assert on line 1884.   The closest one in my source is this line
```
Debug.Assert(eventsToStacks[i].EventIndex < eventsToStacks[i + 1].EventIndex);
```
which would be very suprising because just a few lines higher we sort the eventsToStack array.

**Comment by SudhindraChegu** [comment]
You are right !!!

I am getting AssertionFailed(Lost Condition) from the below mentioned code in TraceLog.cs\CopyRawEvents()

            // The eventsToStacks array is sorted.  
            for (int i = 0; i < eventsToStacks.Count - 1; i++)
                Debug.Assert(eventsToStacks[i].EventIndex < eventsToStacks[i + 1].EventIndex);

**Comment by vancem** [comment]
I need a way of reproducing your problem.  Ideally you put your ETL.ZIP file up on some cloud drive and send me a download link to it.  (vancem@microsoft.com if you wish to do it privately).

**Comment by SudhindraChegu** [comment]
I have sent a mail with the attachment.

**Comment by vancem** [comment]
Note to self: I have downloaded this on vancem1\downloads\PerfViewData.Issue_457.etl

I actually got an 5 asserts  
               Debug.Assert((emitted && IsDead) || KernelModeStackKey != 0); 
               Debug.Assert(eventsToStacks[idx].EventIndex != eventIndex);
               Debug.Assert(UserModeStackIndex == CallStackIndex.Invalid);
               Debug.Assert(UserModeStackIndex >= 0);

But the last ist he one that @SudhindraChegu  saw
                Debug.Assert(eventsToStacks[i].EventIndex < eventsToStacks[i + 1].EventIndex);

Fails when I = 2725.   because the EventIndexes are EQUAL.  Now having two stacks for a given event is also weird, and the previous asserts are probably telling us something about why.

However I think it is fair to say that this asset is likely to indicate that only one stack is 'wrong' and the effect should be confined to one stack.

Thus it is likely to be a minor issue.   You can ignore it for now.  I will see if I can get some time to look at this more carefully, but it may not be for days or more.

**Comment by SudhindraChegu** [comment]
Sure ... Thanks for the analysis !!

**Comment by sharwell** [comment]
:memo: This should be less common now that #424 is merged.

**Comment by brianrob** [comment]
Fixed via #424.
